zefi.py has incorrect assertions

**Describe the bug**
I ran into this build error on up_squared:

```
[1/1] Linking C executable zephyr/zephyr.elf
FAILED: zephyr/zephyr.elf zephyr/zephyr.bin zephyr/zephyr.lst zephyr/zephyr.stat zephyr/zephyr.strip
: && ccache /home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc -m64   zephyr/CMakeFiles/zephyr_final.dir/misc/empty_file.c.obj  -o zephyr/zephyr.elf  zephyr/arch/arch/x86/pagetables.o  zephyr/kobject_hash_renamed.o  -Wl,-T  zephyr/linker_pass_final.cmd  -Wl,-Map=/home/apboie/projects/zephyr2/zephyr/tests/kernel/mem_protect/mem_protect/build/zephyr/zephyr_final.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/arch/x86/core/libarch__x86__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/x86/apollo_lake/libsoc__x86__apollo_lake.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/serial/libdrivers__serial.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/arch/x86/core/offsets/offsets.c.obj  -L"/home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/9.2.0"  -L/home/apboie/projects/zephyr2/zephyr/tests/kernel/mem_protect/mem_protect/build/zephyr  -lgcc  -m64  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/apboie/projects/zephyr2/zephyr/tests/kernel/mem_protect/mem_protect/build/zephyr && /usr/bin/cmake -E rename zephyr_final.map zephyr.map && /home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-objcopy --gap-fill 0xff --output-target=binary --remove-section=.comment --remove-section=COMMON --remove-section=.eh_frame zephyr.elf zephyr.bin && /home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-objdump -d -S zephyr.elf >zephyr.lst && /home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-readelf -e zephyr.elf > zephyr.stat && /home/apboie/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-strip --strip-all zephyr.elf -ozephyr.strip && /usr/bin/python3.8 /home/apboie/projects/zephyr2/zephyr/arch/x86/zefi/zefi.py -f /home/apboie/projects/zephyr2/zephyr/tests/kernel/mem_protect/mem_protect/build/zephyr/zephyr.elf
Traceback (most recent call last):
  File "/home/apboie/projects/zephyr2/zephyr/arch/x86/zefi/zefi.py", line 161, in <module>
    build_elf(args.elf_file)
  File "/home/apboie/projects/zephyr2/zephyr/arch/x86/zefi/zefi.py", line 55, in build_elf
    assert (h.p_filesz % 4) == 0
AssertionError
ninja: build stopped: subcommand failed.
```

This assertion is not correct. From dumping the file we have an unaligned filesiz/memsiz, I have no reason to believe our ELF binary is incorrect:
```
$ readelf -l zephyr/zephyr.elf

Elf file type is EXEC (Executable file)
Entry point 0x104b
There are 3 program headers, starting at offset 64

Program Headers:
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  LOAD           0x0000000000000100 0x0000000000001000 0x0000000000001000
                 0x0000000000006000 0x0000000000006000  RWE    0x20
  LOAD           0x0000000000007000 0x0000000000100000 0x0000000000100000
                 0x000000000001a000 0x00000000044af000  RWE    0x1000
  LOAD           0x0000000000021000 0x00000000045af000 0x00000000045af000
                 0x0000000000405b21 0x0000000000405b21  RW     0x20

 Section to Segment mapping:
  Segment Sections...
   00     .locore
   01     text initlevel app_shmem_regions log_const_sections rodata app_smem bss noinit
   02     datas devices _static_thread_data_area k_timer_area k_heap_area k_mutex_area k_msgq_area k_sem_area pagetables kobject_data
```

**To Reproduce**
Currently I can reproduce this with this PR https://github.com/zephyrproject-rtos/zephyr/pull/28986 and building tests/kernel/mem_protect/mem_protect on up_squared

**Expected behavior**
Build to complete successfully.

**Impact**
Showstopper

--- a/arch/x86/zefi/zefi.c
+++ b/arch/x86/zefi/zefi.c
@@ -72,24 +72,24 @@ uintptr_t __abi efi_entry(void *img_handle, struct efi_system_table *sys_tab)
 	printf("*** Zephyr EFI Loader ***\n");
 
 	for (int i = 0; i < sizeof(zefi_zsegs)/sizeof(zefi_zsegs[0]); i++) {
-		int nwords = zefi_zsegs[i].sz;
-		uint32_t *dst = (uint32_t *)zefi_zsegs[i].addr;
+		int bytes = zefi_zsegs[i].sz;
+		uint8_t *dst = (uint8_t *)zefi_zsegs[i].addr;
 
-		printf("Zeroing %d bytes of memory at %p\n", 4 * nwords, dst);
-		for (int j = 0; j < nwords; j++) {
+		printf("Zeroing %d bytes of memory at %p\n", bytes, dst);
+		for (int j = 0; j < bytes; j++) {
 			dst[j] = 0;
 		}
 	}
 
 	for (int i = 0; i < sizeof(zefi_dsegs)/sizeof(zefi_dsegs[0]); i++) {
-		int nwords = zefi_dsegs[i].sz;
+		int bytes = zefi_dsegs[i].sz;
 		int off = zefi_dsegs[i].off;
-		uint32_t *dst = (uint32_t *)zefi_dsegs[i].addr;
-		uint32_t *src = &((uint32_t *)EXT_DATA_START)[off];
+		uint8_t *dst = (uint8_t *)zefi_dsegs[i].addr;
+		uint8_t *src = &((uint8_t *)EXT_DATA_START)[off];
 
 		printf("Copying %d data bytes to %p from image offset %d\n",
-		       4 * nwords, dst, zefi_dsegs[i].off);
-		for (int j = 0; j < nwords; j++) {
+		       bytes, dst, zefi_dsegs[i].off);
+		for (int j = 0; j < bytes; j++) {
 			dst[j] = src[j];
 		}
 	}

--- a/arch/x86/zefi/zefi.py
+++ b/arch/x86/zefi/zefi.py
@@ -49,24 +49,20 @@ def build_elf(elf_file):
             continue
 
         assert h.p_memsz >= h.p_filesz
-        assert (h.p_vaddr % 8) == 0
-        assert (h.p_filesz % 4) == 0
         assert len(seg.data()) == h.p_filesz
 
         if h.p_filesz > 0:
             sd = seg.data()
             verbose("%d bytes of data at 0x%x, data offset %d"
                 % (len(sd), h.p_vaddr, len(data_blob)))
-            data_segs.append((h.p_vaddr, len(sd) / 4, len(data_blob) / 4))
+            data_segs.append((h.p_vaddr, len(sd), len(data_blob)))
             data_blob = data_blob + sd
 
         if h.p_memsz > h.p_filesz:
             bytesz = h.p_memsz - h.p_filesz
-            if bytesz % 4:
-                bytesz += 4 - (bytesz % 4)
             addr = h.p_vaddr + h.p_filesz
             verbose("%d bytes of zero-fill at 0x%x" % (bytesz, addr))
-            zero_segs.append((addr, bytesz / 4))
+            zero_segs.append((addr, bytesz))
 
     verbose(f"{len(data_blob)} bytes of data to include in image")
 

